diff -r 0bb2eb79e515 Makefile
--- a/Makefile	Mon Dec 16 18:00:04 2013 +0200
+++ b/Makefile	Sat Mar 01 11:49:06 2014 +0100
@@ -42,7 +42,7 @@
 endef
 
 define link
-	$(CC) $(LDFLAGS) $1 $2 $3 -o $@
+	$(CC) $(CFLAGS) $(LDFLAGS) $1 $2 $3 -o $@
 endef
 
 # Builds both GTK+2 and GTK+3 versions of the plugin.
diff -r 0bb2eb79e515 ui.c
--- a/ui.c	Mon Dec 16 18:00:04 2013 +0200
+++ b/ui.c	Sat Mar 01 11:49:06 2014 +0100
@@ -19,6 +19,10 @@
 
 #include "ui.h"
 
+#if !GTK_CHECK_VERSION(2,18,0)
+#define gtk_widget_set_can_focus(widget, canfocus) {if (canfocus) GTK_WIDGET_SET_FLAGS (widget, GTK_CAN_FOCUS); else GTK_WIDGET_UNSET_FLAGS(widget, GTK_CAN_FOCUS);}
+#endif
+
 static GtkWidget *infobar_tabs;
 static GtkWidget *infobar_toggles;
 
