diff --git a/Makefile b/Makefile
index 06bc90e..f9115fe 100644
--- a/Makefile
+++ b/Makefile
@@ -6,11 +6,11 @@ DEPDIR = .deps
 
 UNRARDIR = unrar
 
-CXX = g++
+CXX? = g++
 STRIP = strip
 
-CXXFLAGS = -O2
-CXXFLAGS += -c -fPIC
+CXXFLAGS += -O2
+CXXFLAGS += -fPIC
 CXXFLAGS += -Wall \
 	    -Wno-parentheses \
 	    -Wno-switch \
@@ -23,7 +23,7 @@ ifeq ($(DEBUG),1)
 CXXFLAGS += -g -DDEBUG
 endif
 
-LDFLAGS = -shared -fPIC -pthread
+LDFLAGS = -pthread
 
 BIN = $(APP).so
 
@@ -81,20 +81,20 @@ vpath %.hpp $(UNRARDIR)
 all: $(BIN)
 
 $(BIN): $(addprefix $(OBJDIR)/,$(OBJS))
-	$(CXX) $(LDFLAGS) $^ -o $@
+	$(CXX) -shared $(CXXFLAGS) -o $@ $^ $(CXX_LDFLAGS) $(LDFLAGS)
 ifneq ($(DEBUG),1)
 	$(STRIP) $@
 endif
 
 vpath %.o $(OBJDIR)
 $(OBJDIR)/%.o: %.cpp | $(OBJDIR)
-	$(CXX) $(CXXFLAGS) $< -o $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -c $< -o $@
 $(OBJDIR):
 	-mkdir -p $@
 
 vpath %.d $(DEPDIR)
 $(DEPDIR)/%.d: %.cpp | $(DEPDIR)
-	$(CXX) $(CXXFLAGS) -MM $< > $@
+	$(CXX) $(CXXFLAGS) $(LDFLAGS) -c -MM $< > $@
 	sed -i 's,\($*\)\.o[ :]*,\1.o $*.d: ,g' $@
 $(DEPDIR):
 	-mkdir -p $@
