diff --git a/Makefile b/Makefile
index 78ea084..cdaf06d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,113 +1,70 @@
-APP = vfs_rar
-DEBUG ?= 0
-
-OBJDIR = .objs
-DEPDIR = .deps
-
-UNRARDIR = unrar
-
-CXX = g++
-STRIP = strip
-
-CXXFLAGS = -O2
-CXXFLAGS += -c -fPIC
-CXXFLAGS += -Wall \
-	    -Wno-parentheses \
-	    -Wno-switch \
-	    -Wno-sign-compare \
-	    -Wno-unused-variable -Wno-unused-function \
-	    -Wno-maybe-uninitialized
-CXXFLAGS += -I. -I./unrar
-CXXFLAGS += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -DRAR_SMP
-ifeq ($(DEBUG),1)
-CXXFLAGS += -g -DDEBUG
-endif
-
-LDFLAGS = -shared -fPIC -pthread
-
-BIN = $(APP).so
-
-UNRAR_OBJS = filestr.o recvol.o rs.o scantree.o qopen.o
-UNRAR_OBJS += strlist.o \
-	      strfn.o \
-	      pathfn.o \
-	      smallfn.o \
-	      global.o \
-	      file.o \
-	      filefn.o \
-	      filcreat.o \
-	      archive.o \
-	      arcread.o \
-	      unicode.o \
-	      system.o \
-	      isnt.o \
-	      crypt.o \
-	      crc.o \
-	      rawread.o \
-	      encname.o \
-	      resource.o \
-	      match.o \
-	      timefn.o \
-	      rdwrfn.o \
-	      consio.o \
-	      options.o \
-	      errhnd.o \
-	      rarvm.o \
-	      secpassword.o \
-	      rijndael.o \
-	      getbits.o \
-	      sha1.o \
-	      sha256.o \
-	      blake2s.o \
-	      hash.o \
-	      extinfo.o \
-	      extract.o \
-	      volume.o \
-	      list.o \
-	      find.o \
-	      unpack.o \
-	      headers.o \
-	      threadpool.o \
-	      rs16.o \
-	      cmddata.o
-
-OBJS = $(UNRAR_OBJS) $(APP).o
-
-vpath %.cpp $(UNRARDIR)
-vpath %.hpp $(UNRARDIR)
-
-.PHONY: all
-all: $(BIN)
-
-$(BIN): $(addprefix $(OBJDIR)/,$(OBJS))
-	$(CXX) $(LDFLAGS) $^ -o $@
-ifneq ($(DEBUG),1)
-	$(STRIP) $@
-endif
-
-vpath %.o $(OBJDIR)
-$(OBJDIR)/%.o: %.cpp | $(OBJDIR)
-	$(CXX) $(CXXFLAGS) $< -o $@
-$(OBJDIR):
-	-mkdir -p $@
-
-vpath %.d $(DEPDIR)
-$(DEPDIR)/%.d: %.cpp | $(DEPDIR)
-	$(CXX) $(CXXFLAGS) -MM $< > $@
-	sed -i 's,\($*\)\.o[ :]*,\1.o $*.d: ,g' $@
-$(DEPDIR):
-	-mkdir -p $@
-
--include $(addprefix $(DEPDIR)/,$(OBJS:.o=.d))
-
-.PHONY: install uninstall clean
-install: $(BIN)
-	-mkdir -p ~/.local/lib/deadbeef
-	-cp $(BIN) ~/.local/lib/deadbeef
-
-uninstall:
-	-rm -rf ~/.local/lib/deadbeef/$(BIN)
+OUT=vfs_rar.so
+CXX?=g++
+CXXFLAGS+=-D_GNU_SOURCE -I./ -I./unrar -DSILENT -fPIC
+
+LDFLAGS+=-shared
+
+RARSOURCES=\
+	unrar/strlist.cpp \
+	unrar/strfn.cpp \
+	unrar/pathfn.cpp \
+	unrar/smallfn.cpp \
+	unrar/global.cpp \
+	unrar/file.cpp \
+	unrar/filefn.cpp \
+	unrar/filcreat.cpp \
+	unrar/archive.cpp \
+	unrar/arcread.cpp \
+	unrar/unicode.cpp \
+	unrar/system.cpp \
+	unrar/isnt.cpp \
+	unrar/crypt.cpp \
+	unrar/crc.cpp \
+	unrar/rawread.cpp \
+	unrar/encname.cpp \
+	unrar/resource.cpp \
+	unrar/match.cpp \
+	unrar/timefn.cpp \
+	unrar/rdwrfn.cpp \
+	unrar/consio.cpp \
+	unrar/options.cpp \
+	unrar/errhnd.cpp \
+	unrar/rarvm.cpp \
+	unrar/secpassword.cpp \
+	unrar/rijndael.cpp \
+	unrar/getbits.cpp \
+	unrar/sha1.cpp \
+	unrar/extinfo.cpp \
+	unrar/extract.cpp \
+	unrar/volume.cpp \
+	unrar/list.cpp \
+	unrar/find.cpp \
+	unrar/unpack.cpp \
+	unrar/cmddata.cpp \
+	unrar/recvol.cpp\
+	unrar/scantree.cpp\
+	unrar/rs.cpp\
+	unrar/filestr.cpp\
+	unrar/qopen.cpp\
+	unrar/rs16.cpp\
+	unrar/hash.cpp\
+	unrar/blake2s.cpp\
+	unrar/sha256.cpp\
+	unrar/headers.cpp
+
+SOURCES=vfs_rar.cpp $(RARSOURCES)
+
+OBJECTS=$(SOURCES:.cpp=.o)
+
+all: $(SOURCES) $(OUT)
+
+$(OUT): $(OBJECTS)
+	$(CXX) $(OBJECTS) $(CXXFLAGS) $(LDFLAGS) -o $@
+
+.c.o:
+	$(CXX) $(CFLAGS) $(CXXFLAGS) $< -c -o $@
 
 clean:
-	-rm -rf $(BIN) $(OBJDIR) $(DEPDIR)
+	rm -f $(OBJECTS) $(OUT)
+
 
