diff --git a/Makefile.am b/Makefile.am
index 75040c5..b3a1dba 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -36,13 +36,13 @@ endif
 # normal lib
 if HAVE_GTK2
 ddb_misc_filebrowser_GTK2_la_SOURCES = $(filebrowser_SOURCES)
-ddb_misc_filebrowser_GTK2_la_LDFLAGS = -module
+ddb_misc_filebrowser_GTK2_la_LDFLAGS = -module -avoid-version
 ddb_misc_filebrowser_GTK2_la_LIBADD  = $(LDADD) $(GTK2_DEPS_LIBS)
 ddb_misc_filebrowser_GTK2_la_CFLAGS  = -std=c99 $(GTK2_DEPS_CFLAGS) -Wall -Werror -g
 endif
 if HAVE_GTK3
 ddb_misc_filebrowser_GTK3_la_SOURCES = $(filebrowser_SOURCES)
-ddb_misc_filebrowser_GTK3_la_LDFLAGS = -module
+ddb_misc_filebrowser_GTK3_la_LDFLAGS = -module -avoid-version
 ddb_misc_filebrowser_GTK3_la_LIBADD  = $(LDADD) $(GTK3_DEPS_LIBS)
 ddb_misc_filebrowser_GTK3_la_CFLAGS  = -std=c99 $(GTK3_DEPS_CFLAGS) -Wall -Werror -g
 endif
diff --git a/filebrowser.c b/filebrowser.c
index 91c0dbd..2cc9f59 100644
--- a/filebrowser.c
+++ b/filebrowser.c
@@ -2083,7 +2083,7 @@ on_treeview_mousemove (GtkWidget *widget, GdkEventButton *event)
                 .info = 0
             };
             GtkTargetList *target = gtk_target_list_new (&entry, 1);
-#if !GTK_CHECK_VERSION(3,0,0)
+#if !GTK_CHECK_VERSION(3,10,0)
             gtk_drag_begin (widget, target, GDK_ACTION_COPY | GDK_ACTION_MOVE, 1, (GdkEvent *)event);
 #else
             gtk_drag_begin_with_coordinates (widget, target, GDK_ACTION_COPY | GDK_ACTION_MOVE, 1, (GdkEvent *)event, -1, -1);
