--- a/Makefile.am.org	2015-03-04 21:33:22.400353452 +0300
+++ b/Makefile.am	2015-03-04 21:33:51.020581376 +0300
@@ -40,15 +40,15 @@
 
 if HAVE_GTK2
 ddb_misc_filebrowser_GTK2_la_SOURCES = $(filebrowser_SOURCES)
-ddb_misc_filebrowser_GTK2_la_LDFLAGS = -module
+ddb_misc_filebrowser_GTK2_la_LDFLAGS = -module -avoid-version
 ddb_misc_filebrowser_GTK2_la_LIBADD  = $(LDADD) $(GTK2_DEPS_LIBS)
-ddb_misc_filebrowser_GTK2_la_CFLAGS  = -std=c99 $(GTK2_DEPS_CFLAGS) -Wall -Werror $(DEBUG_FLAGS)
+ddb_misc_filebrowser_GTK2_la_CFLAGS  = -std=c99 $(GTK2_DEPS_CFLAGS) -Wall $(DEBUG_FLAGS)
 endif
 if HAVE_GTK3
 ddb_misc_filebrowser_GTK3_la_SOURCES = $(filebrowser_SOURCES)
-ddb_misc_filebrowser_GTK3_la_LDFLAGS = -module
+ddb_misc_filebrowser_GTK3_la_LDFLAGS = -module -avoid-version
 ddb_misc_filebrowser_GTK3_la_LIBADD  = $(LDADD) $(GTK3_DEPS_LIBS)
-ddb_misc_filebrowser_GTK3_la_CFLAGS  = -std=c99 $(GTK3_DEPS_CFLAGS) -Wall -Werror $(DEBUG_FLAGS)
+ddb_misc_filebrowser_GTK3_la_CFLAGS  = -std=c99 $(GTK3_DEPS_CFLAGS) -Wall $(DEBUG_FLAGS)
 endif
 
 docsdir = $(docdir)
