diff --git a/utils.c b/utils.c
index b687ab3..a3d3bf0 100644
--- a/utils.c
+++ b/utils.c
@@ -234,7 +234,11 @@ utils_check_dir (const gchar *dir, mode_t mode)
         if (slash)
             *slash = 0;
         if (-1 == stat (tmp, &stat_buf)) {
+#ifdef __MINGW32__
+            int err_no = mkdir (tmp);
+#else
             int err_no = mkdir (tmp, mode);
+#endif
             if (0 != err_no) {
                 fprintf (stderr, "Failed to create %s (%d)\n", tmp, err_no);
                 g_free (tmp);
