diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..3830bf5
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,13 @@
+CC?=gcc
+GLIB_CFLAGS?=`pkg-config --cflags dbus-glib-1 dbus-1 glib-2.0`
+GLIB_LIBS?=`pkg-config --libs dbus-glib-1 dbus-1 glib-2.0`
+LDFLAGS+=-std=c99 -shared -O2
+NAME='ddb_pause_on_lock'
+
+all:
+	echo 'VOID:BOOLEAN' >${NAME}_marshal.list
+	glib-genmarshal --header --prefix=marshal ${NAME}_marshal.list > ${NAME}_marshal.h
+	$(CC) $(CFLAGS) $(GLIB_CFLAGS) $(GLIB_LIBS) $(LDFLAGS) -o ddb_pause_on_lock.so ddb_pause_on_lock.c
+
+clean:
+	rm ddb_pause_on_lock.so
