diff --git a/Makefile b/Makefile
index 76da33f..629d066 100755
--- a/Makefile
+++ b/Makefile
@@ -39,12 +39,12 @@ ARTWORK_OBJS=artwork/artwork_internal.o artwork/escape.o artwork/lastfm.o
 all: submodules_load libdiscord-rpc.a discord_presence
 
 discord_presence:
-	$(MAKE) -C artwork
+	CFLAGS="$(CFLAGS)" $(MAKE) -C artwork
 	$(CC) -std=$(STD) -c $(CFLAGS) -c $(PLUGNAME).c
 	$(CXX) -std=$(STD) -shared $(CXXFLAGS) -o $(PLUGNAME).$(SUFFIX) $(PLUGNAME).o $(ARTWORK_OBJS) $(LIBS) $(CXX_LDFLAGS) $(LDFLAGS)
 
 libdiscord-rpc.a: discord-rpc-patch
-	cd discord-rpc && $(MAKE)
+	cd discord-rpc && CFLAGS="$(CFLAGS)" $(MAKE)
 	cp discord-rpc/src/libdiscord-rpc.a .
 
 submodules_load:
@@ -71,3 +71,4 @@ install:
 clean: discord-rpc-patch-reverse
 	cd discord-rpc && git clean -df && git reset --hard
 	rm -fv $(PLUGNAME).o $(PLUGNAME).$(SUFFIX)
+	rm artwork/*.o
