diff --git a/Makefile b/Makefile
new file mode 100644
index 0000000..f2f65b0
--- /dev/null
+++ b/Makefile
@@ -0,0 +1,2 @@
+all:
+	$(CC) $(CFLAGS) -std=c99 -fPIC $(LDFLAGS) -shared -o rating.so rating.c
diff --git a/rating.c b/rating.c
index 98f70d2..b4f9f61 100644
--- a/rating.c
+++ b/rating.c
@@ -1,4 +1,4 @@
-#include <deadbeef.h>
+#include <deadbeef/deadbeef.h>
 #include <string.h>
 #include <stdlib.h>
 
@@ -220,8 +220,8 @@ static DB_misc_t plugin = {
     .plugin.version_major = 1,
     .plugin.version_minor = 1,
     .plugin.type = DB_PLUGIN_MISC,
-    .plugin.name = "rating",
-    .plugin.descr = "Enables commands to rate song(s) by editing the metadata tag rating.",
+    .plugin.name = "Song Rating",
+    .plugin.descr = "Adds menu commands for rating songs, reads/writes the ratings from/to file tags.",
     .plugin.copyright =
     "Rating plugin for DeaDBeeF Player\n"
     "Author: Christian Hernvall\n"
@@ -240,7 +240,7 @@ static DB_misc_t plugin = {
     "along with this program; if not, write to the Free Software\n"
     "Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.\n"
     ,
-    .plugin.website = "http://deadbeef.sf.net",
+    .plugin.website = "https://github.com/splushii/deadbeef-rating",
     .plugin.start = rating_start,
     .plugin.stop = rating_stop,
     .plugin.get_actions = rating_get_actions,
