diff --git a/Makefile b/Makefile
index 0be786c..0149e50 100644
--- a/Makefile
+++ b/Makefile
@@ -5,13 +5,13 @@ INCLUDE=-I${PREFIX}/include -I${PREFIX}/include/opus -I${PREFIX}/include/deadbee
 CC?=gcc
 
 CFLAGS+=-fPIC -Wall -Wstrict-overflow=5 -fstrict-overflow -Wno-sequence-point -Wno-parentheses -g -Os -std=c99 -fPIC -D_GNU_SOURCE
-CFLAGS+=-DHAVE_OGG_STREAM_FLUSH_FILL # ogg_stream_flush_fill() function is available (libogg v1.3.0 onwards)
+# CFLAGS+=-DHAVE_OGG_STREAM_FLUSH_FILL # ogg_stream_flush_fill() function is available (libogg v1.3.0 onwards)
 CFLAGS+=-DUSE_OPUSFILE_FLOAT # decode to floats (disable for pure fixed point decoding dithered to 16-bit ints)
-CFLAGS+=-DUSE_OPUSURL # to use libopusurl.so for fetching streaming station data
+# CFLAGS+=-DUSE_OPUSURL # to use libopusurl.so for fetching streaming station data
 # CFLAGS+=-D_FILE_OFFSET_BITS=64 # for largefile support on 32 bit systems
 CFLAGS+=${INCLUDE}
 
-LDFLAGS+= -s -shared -lopusfile -Wl,--exclude-libs=ALL
+LDFLAGS+= -s -shared -lopusfile -logg -lopus -lm -Wl,--exclude-libs=ALL
 ifeq ($(USE_OPUSURL), true)
 	LDFLAGS+=-lopusurl
 endif
@@ -23,7 +23,7 @@ OBJECTS=$(SOURCES:.c=.o)
 all: $(SOURCES) $(OUT)
 
 $(OUT): $(OBJECTS)
-	$(CC) $(OBJECTS) $(LDFLAGS) -o $@
+	$(CC) $(CFLAGS) $(OBJECTS) $(LDFLAGS) -o $@
 
 .c.o:
 	$(CC) $(CFLAGS) $< -c -o $@
