diff --git a/vumeter.c b/vumeter.c
index 81e7b82..773dabc 100644
--- a/vumeter.c
+++ b/vumeter.c
@@ -808,7 +808,7 @@ vumeter_set_refresh_interval (gpointer user_data, int interval)
 }
 
 static void
-vumeter_wavedata_listener (void *ctx, ddb_audio_data_t *data) {
+vumeter_wavedata_listener (void *ctx, const ddb_audio_data_t *data) {
     w_vumeter_t *w = ctx;
     deadbeef->mutex_lock (w->mutex);
     w->channels = MIN (MAX_CHANNELS, data->fmt->channels);
@@ -879,7 +879,7 @@ vumeter_draw_bars (w_vumeter_t *w, cairo_t *cr, int width, int height)
 
     unsigned char *data = cairo_image_surface_get_data (w->surf);
     if (!data) {
-        return FALSE;
+        return;
     }
     int stride = cairo_image_surface_get_stride (w->surf);
     memset (data, 0, height * stride);
