diff --git a/Makefile.am b/Makefile.am
index 42a5ed8..9aaad2f 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,10 +5,8 @@ mpris_la_SOURCES = mpris.c \
                    introspection_xml.h introspection_xml.c \
                    mpris_common.h mpris_common.c
 
-mpris_la_LDFLAGS = -module
+mpris_la_LDFLAGS = -module -avoid-version
 
-mpris_la_LIBADD = $(LDADD) $(MPRIS_LIBS)
-AM_CFLAGS = $(CFLAGS) $(MPRIS_CFLAGS) -std=c99 -Wall -Werror \
-                -I/usr/local/include
+mpris_la_LIBADD = $(LDADD) $(MPRIS_DEPS_LIBS)
+AM_CFLAGS = $(CFLAGS) $(MPRIS_DEPS_CFLAGS) -std=c99 -Wall -Werror
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS=test
diff --git a/configure.ac b/configure.ac
index 32e8cb9..397b6e6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -15,13 +15,7 @@ LT_INIT
 AC_C_BIGENDIAN
 
 PKG_PROG_PKG_CONFIG
-PKG_CHECK_MODULES(GLIB_DEPS, glib-2.0)
-PKG_CHECK_MODULES(GIO_DEPS, gio-2.0)
-PKG_CHECK_MODULES(GTHREAD_DEPS, gthread-2.0)
-MPRIS_LIBS="$GLIB_DEPS_LIBS $GIO_DEPS_LIBS $GTHREAD_DEPS_LIBS"
-MPRIS_CFLAGS="$GLIB_DEPS_CFLAGS $GIO_DEPS_CFLAGS $GTHREAD_DEPS_CFLAGS"
-AC_SUBST(MPRIS_LIBS)
-AC_SUBST(MPRIS_CFLAGS)
+PKG_CHECK_MODULES(MPRIS_DEPS, glib-2.0 gio-2.0 gthread-2.0)
 
 dnl Search for deadbeef installation directory
 AC_PREFIX_PROGRAM(deadbeef)
