diff --git a/src/headerbarui.c b/src/headerbarui.c
index 76f0bf2..026249b 100644
--- a/src/headerbarui.c
+++ b/src/headerbarui.c
@@ -1144,7 +1144,7 @@ refresh_undo_redo_menu () {
 }
 
 static void init_refresh_undo_redo_menu () {
-    GActionGroup *db_action_group = gtk_widget_get_action_group (headerbar, "db");
+    GActionGroup *db_action_group = G_ACTION_GROUP(gtk_widget_get_action_group (headerbar, "db"));
 
     // Use the menu items
     refresh_undo_redo.undo_menu_item = lookup_widget (GTK_WIDGET(mainwin), "undo");
@@ -1195,10 +1195,10 @@ void window_init_hook (void *userdata) {
     headerbar_add_menu_btn = GTK_BUILDER_GET_WIDGET(builder, "file_menu_btn");
     headerbar_playback_menu_btn = GTK_BUILDER_GET_WIDGET(builder, "playback_menu_btn");
 
-    GActionGroup *group = create_action_group();
+    GActionGroup *group = G_ACTION_GROUP(create_action_group());
     gtk_widget_insert_action_group (headerbar, "win", group);
 
-    GActionGroup *deadbeef_action_group = create_action_group_deadbeef();    
+    GActionGroup *deadbeef_action_group = G_ACTION_GROUP(create_action_group_deadbeef());    
     gtk_widget_insert_action_group (headerbar, "db", deadbeef_action_group);
 
     update_plugin_actions();
